scripts/build.sh

```diff
@@ -1,50 +1,28 @@
 #!/usr/bin/env bash
+set -e -u -o pipefail
 
-# This is a generic build script for the GDK for Unity.
-# The expected usage is as follows:
-#   bash build.sh <unity_project_dir> <worker_type> <build_target> <log_file (optional)>
+if [[ -n "${DEBUG-}" ]]; then
+  set -x
+fi
 
-set -e -u -x -o pipefail
-
-UNITY_PROJECT_DIR=$1
-WORKER_TYPE=$2
-BUILD_TARGET=$3
-SCRIPTING_BACKEND=$4
-LOG_FILE=${5:-} # Optional argument - will log to console otherwise
-
-# The asset cache ip cannot be hardcoded and so is stored in an environment variable on the build agent.
-# This is bash shorthand syntax for if-else predicated on the existance of the environment variable 
-# where the else branch assigns an empty string.
-#   i.e. - 
-#   if [ -z ${UNITY_ASSET_CACHE_IP} ]; then
-#       ASSET_CACHE_ARG="-CacheServerIPAddress ${UNITY_ASSET_CACHE_IP}"
-#   else
-#       ASSET_CACHE_ARG=""
-#   fi
-
-ASSET_CACHE_ARG=${UNITY_ASSET_CACHE_IP:+-CacheServerIPAddress "${UNITY_ASSET_CACHE_IP}"}
+echo "Building for: ${WORKER_TYPE} ${BUILD_TARGET} ${SCRIPTING_TYPE}"
 
 pushd "$(dirname "$0")/../"
-    source "scripts/pinned-tools.sh"
-    source "scripts/profiling.sh"
+  source "scripts/pinned-tools.sh"
 
-    RUN_UNITY_PATH="$(pwd)/tools/RunUnity/RunUnity.csproj"
-popd
+  if [[ ${WORKER_TYPE} == "AndroidClient" ]]; then
+      scripts/prepare-unity-mobile.sh "$(pwd)/../logs/PrepareUnityMobile.log"
+  fi
 
-markStartOfBlock "Building ${WORKER_TYPE} for ${BUILD_TARGET} and ${SCRIPTING_BACKEND}"
+  if [[ ${WORKER_TYPE} == "iOSClient" ]]; then
+      if ! isMacOS; then
+          echo "I can't build for iOS!"
+          exit 0
```

Comment:
do we expect this to happen on non-macos machines?

Reply:
Yes - this is just a guard against misconfiguration, in which case this may want to exit 1? 

I've left the build iOS on the steps list and this early outs - once we have Mac Buildkite we can target that step at the Mac agent

Reply:
yeah, I think we should exit 1 here to make it clear that something got misconfigured